src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx

```diff
@@ -1072,7 +1072,9 @@ export default function EnvTriggerView({ filteredAppIds, isVirtualEnv }: AppGrou
                             isSelected: i === index,
                         }
                     })
-                    setFilterMaterials(artifacts)
+                    if (filterMaterials.length > 0) {
+                        setFilterMaterials(artifacts)
+                    }
```

Comment:
Can we add additional approval node check if possible to be more specific? As this will only be used at approval node modal.